src/plug/MainView.vala

```diff
@@ -5,127 +5,55 @@
  */
 
 public class PC.MainBox : Gtk.Box {
-    private List<Widgets.UserItem> items;
-    private Gtk.ListBox list;
+    private Gtk.Stack stack;
 
     construct {
-        var stack = new Gtk.Stack () {
-            hexpand = true
-        };
-
-        items = new List<Widgets.UserItem> ();
+        stack = new Gtk.Stack ();
 
-        list = new Gtk.ListBox () {
-            selection_mode = SINGLE
-        };
-        list.set_header_func (update_headers);
-
-        var scrolled_window = new Gtk.ScrolledWindow () {
-            child = list,
-            hscrollbar_policy = NEVER,
-            vexpand = true
-        };
+        var sidebar = new Switchboard.SettingsSidebar (stack);
 
         var paned = new Gtk.Paned (HORIZONTAL) {
-            position = 240,
-            start_child = scrolled_window,
+            start_child = sidebar,
             end_child = stack,
             shrink_start_child = false,
             shrink_end_child = false,
-            resize_start_child = false
+            resize_start_child = false,
+            vexpand = true
         };
 
         orientation = VERTICAL;
         append (paned);
 
-        list.row_activated.connect ((row) => {
-            var user_item_row = (Widgets.UserItem) row;
-            if (stack.get_child_by_name (user_item_row.user.uid.to_string ()) == null) {
-                stack.add_named (user_item_row.page, user_item_row.user.uid.to_string ());
-            }
-
-            stack.visible_child = user_item_row.page;
-        });
-
         unowned var user_manager = Act.UserManager.get_default ();
         user_manager.user_added.connect (add_user);
         user_manager.user_changed.connect (update_user);
         user_manager.user_removed.connect (remove_user);
 
         if (user_manager.is_loaded) {
             user_manager.list_users ().foreach ((user) => add_user (user));
-            select_first ();
         } else {
             user_manager.notify["is-loaded"].connect (() => {
                 user_manager.list_users ().foreach ((user) => add_user (user));
-                select_first ();
             });
         }
     }
 
     private void add_user (Act.User user) {
-        if (has_user (user)) {
+        if (stack.get_child_by_name (user.uid.to_string ()) != null) {
             return;
         }
 
         var page = new Widgets.ControlPage (user);
-        var useritem = new Widgets.UserItem (page);
-
-        items.append (useritem);
-        list.append (useritem);
-
-        select_first ();
+        stack.add_titled (page, user.uid.to_string (), page.title);
     }
 
     private void update_user (Act.User user) {
-        foreach (var item in items) {
-            if (item.user == user) {
-                item.update_view ();
-                add_user (user);
-
-                break;
-            }
-        }
+        ((Widgets.ControlPage) stack.get_child_by_name (user.uid.to_string ())).update_user;
```

Comment:
```suggestion
        ((Widgets.ControlPage) stack.get_child_by_name (user.uid.to_string ())).update_user ();
```

I'm not quite sure what it does otherwise?